src/mainboard/pcengines/apu2/mainboard.c

```diff
@@ -250,6 +250,34 @@ static void mainboard_enable(struct device *dev)
 
 static void mainboard_final(void *chip_info)
 {
+	//
+	// AGESA set bit 0, EccExclEn, in register D18F5x240 as 1.
+	// Range was incorrectly enabled to cover all memory
+	// for cases without UMA (no integrated graphics).
+	//
+	// In order to clear EccExclEn DRAM scrubber needs to be
+	// disabled temporarily by setting D18F3x88[DisDramScrub]
+	//
+	u32 val;
+	device_t D18F3 = dev_find_device(0x1022, 0x1583, NULL);
+	device_t D18F5 = dev_find_device(0x1022, 0x1585, NULL);
+
+	val = pci_read_config32(D18F3, 0x88);
+	val |= (1 << 27);		// Disable DRAM ECC scrubbing
+	pci_write_config32(D18F3, 0x88, val);
+
+	val = pci_read_config32(D18F3, 0xB8);
+	val &= 0xF000003F;		// Reserved bits, but injection doesn't work if set
```

Comment:
@krystian-hebel probably you should add not that this was figured out by experimentation.